common/djangoapps/dark_lang/views.py

```diff
@@ -0,0 +1,153 @@
+
+    Expected Behavior:
+    GET - returns a form for setting/resetting the user's dark language
+    POST - updates or clears the setting to the given dark language
+
+    Note: This class and View are meant to replace the ability to set the DarkLang in the middleware, which was
+    determined to be a security risk. (TNL-4742)
+    """
+    template_name = 'darklang/preview_lang.html'
+
+    @method_decorator(login_required)
+    def get(self, request):
+        """
+        Displays the Form for setting/resetting a User's dark language setting
+
+        Arguments:
+            request (Request): The Django Request Object
+
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_pattern_library': True
+        }
+        return render_to_response(self.template_name, context)
+
+    @method_decorator(login_required)
+    def post(self, request):
+        """
+        Sets or clears the DarkLang depending on the incoming post data.
+
+        Arguments:
+            request (Request): The Django Request Object
+
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_pattern_library': True
+        }
+        (success, result) = self.process_darklang_request(request)
+        if result is not None:
+            context.update({'form_submit_message': result})
+            context.update({'success': success})
+        return render_to_response(self.template_name, context)
+
+    def process_darklang_request(self, request):
+        """
+        Prevent user from requesting un-released languages
+        """
+        if not DarkLangConfig.current().enabled:
+            return False, _('Preview Language is currently disabled')
+
+        success = False
+        result = None
+        if 'reset' in request.POST:
+            # Reset and clear the language preference
+            (success, result) = self._clear_preview_language(request)
+        if 'set_language' in request.POST:
+            # Set the Preview Language
+            (success, result) = self._set_preview_language(request)
+        return success, result
+
+    def _set_preview_language(self, request):
+        """
+        Attempt to set the Preview language
+
+        Arguments:
+            request (Request): The incoming Django Request
+
+        Returns:
+            bool: True for successful setting of the language code
+            str: The response message to be presented
+        """
+        if LANGUAGE_INPUT_FIELD not in request.POST:
+            return False, _('Language code not provided')
+
+        preview_lang = request.POST[LANGUAGE_INPUT_FIELD]
+
+        if preview_lang == '':
+            return False, _('Language code not provided')
+
+        auth_user = request.user.is_authenticated()
```

Comment:
Question: Why is it necessary to call `is_authenticated` instead of just using `request.user`?  The `login_required` decorator already verifies that the user is logged in and authenticated: https://github.com/django/django/blob/master/django/contrib/auth/decorators.py#L46